merkle-tree/src/lib.rs

```diff
@@ -1 +1,60 @@
+use serde::{Deserialize, Serialize};
+use simperby_common::crypto::Hash256;
+use simperby_common::MerkleProof;
+use simperby_kv_storage::KVStorage;
+use std::marker::PhantomData;
+use thiserror::Error;
 
+#[derive(Error, Debug, Serialize, Deserialize, Clone)]
+pub enum Error {
+    /// When the given storage is not a valid Merkle-tree-indexed database.
+    #[error("invalid storage {0}")]
+    MalformedStorage(String),
+    /// When the provided storage emitted an error during the operation.
+    #[error("storage error: {0}")]
+    StorageError(simperby_kv_storage::Error),
+}
+
+/// A state machine that interpret the given storage as a Merkle tree rooted by a specific key.
+///
+/// Note that there is no `get()` method as leaf nodes (which the user of this struct would like to read)
```

Comment:
I don't understand this part. Isn't it that the `MerkleTree` is an upper-level abstraction upon `KVstorage`? In that case, I believe `get()` method is still needed unless our PBC code holds both `KVstorage` and `MerkleTree`.

Reply:
No it's not. MerkleTree is just one way of interpreting the given storage. We still need to access to the underlying storage for the following cases.
1. We need to access to entries that are not part of the Merkle tree. (e.g., past blocks)
2. We need to use the 'checkpoint' functionalities provided by the storage.
That makes it natural not to provide `get()` method that would just invoke another `get()` of the underlying storage.

Reply:
I add a commit about this. Please check https://github.com/postech-dao/simperby/pull/39/commits/a6d9a653000896de8a3fd56efbb5eaf78c222c54